app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -24,21 +24,38 @@ function checkIfNearBottomOfPage() {
 }
 
 function fetchNextPageIfNearBottom() {
-  var elCheck = document.getElementById("index-container") && !document.getElementById("query-wrapper");
+  var indexContainer = document.getElementById("index-container");
+  var elCheck =  indexContainer && !document.getElementById("query-wrapper");
   if (!elCheck) {
     return;
   }
-  if ( !done && !fetching && (window.scrollY) > (document.documentElement.scrollHeight - 3700) ) {
+  
+  var indexWhich = indexContainer.dataset.which;
+
+  var fetchCallback;
+  var scrollableElemId;
+  if (indexWhich == "podcast-episodes") {
+    scrollableElemId = "articles-list";
+    fetchCallback = function() {
+      fetchNextPodcastPage(indexContainer);
+    };
+  } else if (indexWhich == "videos") {
+    scrollableElemId = "video-collection";
+    fetchCallback = function() {
+      fetchNextVideoPage(indexContainer);
+    };
+  } else {
+    scrollableElemId = "articles-list";
+    fetchCallback = function() {
+      algoliaPaginate(indexContainer.dataset.algoliaTag);
+    };
+  }
+
+  var scrollableElem = document.getElementById(scrollableElemId);
+
+  if ( !done && !fetching && (window.scrollY) > (scrollableElem.scrollHeight - 3700) ) {
```

Comment:
This line is the meat - previously we were using the document element in this calculation. That's a problem when the sidebar has a considerable amount of content. When the height of the sidebar reaches a certain threshold, it impacts this calculation.